torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1319,6 +1319,35 @@ def _wait_all_workers(self, f, x):
         dist.barrier()
         rpc.shutdown(graceful=False)
 
+    @dist_init(setup_rpc=False)
+    def test_wait_all_workers_timeout(self):
+        initialize_pg(self.file_init_method, self.rank, self.world_size)
+
+        rpc.init_rpc(
+            name=worker_name(self.rank),
+            backend=self.rpc_backend,
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_backend_options=self.rpc_backend_options,
+        )
+
+        og_func = rpc.api._wait_all_workers
+
+        def wait_all_workers_sleep(timeout):
```

Comment:
This is a pretty clever way to test it! I was originally thinking about sth like letting rank0 using an self-RPC to occupy its thread to trigger the timeout. I think yours is better, and less depend on the RPC threading model.